File: python/helpers/IOUtils.py

#serve para validar um input de int e 'levanta' o erro
def validar_input(prompt: str = "") -> int | Exception:
    while(True):
        try:
            return int(input(prompt))
        except ValueError as e:
            #erro a nível de sistema, preferência ao inglês.
            e.args = "Int conversion failed - user likely inputted a string. "

            print("Favor digitar um número.")
            continue

File: java/helpers/IOUtils.java
package helpers;

public class IOUtils {

    // Valida um input de int, repetindo até receber um número válido.
    public static int validarInput(String prompt) {
        while (true) {
            System.out.print(prompt);
            String line = IO.readln();
            try {
                return Integer.parseInt(line.trim());
            } catch (NumberFormatException e) {
                System.out.println("Favor digitar um número.");
            }
        }
    }
}
